model/src/main/resources/db-update/30200-build-push.sql

```diff
@@ -0,0 +1,160 @@
+--  buildrecord_id             | bigint                 |           |          |
+--  productmilestonerelease_id | bigint                 |           |          |
+--  userinitiator              | character varying(255) |           |          |
+
+
+-- New tables
+
+--                             Table "operation"
+--        Column        |           Type           | Collation | Nullable | Default
+-- ---------------------+--------------------------+-----------+----------+---------
+--  operation_type      | character varying(31)    |           | not null |
+--  id                  | bigint                   |           | not null |
+--  endtime             | timestamp with time zone |           |          |
+--  progressstatus      | character varying(255)   |           |          |
+--  result              | character varying(255)   |           |          |
+--  starttime           | timestamp with time zone |           |          |
+--  submittime          | timestamp with time zone |           |          |
+--  user_id             | integer                  |           | not null |
+--  productmilestone_id | integer                  |           |          |
+--  build_id            | bigint                   |           |          |
+
+--                    Table "operation_parameters"
+--     Column    |          Type           | Collation | Nullable | Default
+-- --------------+-------------------------+-----------+----------+---------
+--  operation_id | bigint                  |           | not null |
+--  value        | character varying(8192) |           | not null |
+--  key          | character varying(50)   |           | not null |
+
+--                     Table "buildpushreport"
+--    Column    |          Type          | Collation | Nullable | Default
+-- -------------+------------------------+-----------+----------+---------
+-- brewbuildid  | integer                |           | not null |
+-- brewbuildurl | character varying(255) |           |          |
+-- operation_id | bigint                 |           | not null |
+
+
+BEGIN; -- New and updated tables
+
+    ALTER table operation ADD build_id bigint;
+    ALTER TABLE operation
+        ADD CONSTRAINT fk_operation_buildrecord FOREIGN KEY (build_id) REFERENCES buildrecord(id);
+
+    CREATE TABLE buildpushreport
+    (
+        brewbuildid  integer NOT NULL,
+        brewbuildurl character varying(255),
+        operation_id bigint  NOT NULL
+    );
+    ALTER TABLE buildpushreport
+        ADD CONSTRAINT buildpushreport_pkey PRIMARY KEY (operation_id);
+    ALTER TABLE buildpushreport
+        ADD CONSTRAINT fk_buildpushreport_operation FOREIGN KEY (operation_id) REFERENCES operation(id);
+
+COMMIT;
+
+BEGIN; -- Data migration
+
+    INSERT INTO operation
+    SELECT
+        'BuildPush' AS operation_type,
+        bpr.id AS id,
+        to_timestamp(((bpr.id >> 22) + 1577836800000 + 1800000)::double precision/1000) AS endtime,
```

Comment:
maybe you can simplify this with:
`to_timestamp((bpr.id >> 22) / 1000 + 1577836800 + 1800) AS endtime`

Reply:
I need the `::double precision` to keep milisecods and this works so I won't touch it :D